src/cmd/actions.rs

```diff
@@ -0,0 +1,63 @@
+/// Represents actions that are available while reading live output from a process
+pub struct Actions {
+    lines: Vec<String>,
```

Comment:
This struct will be used in the hot path of rustwide, and I'd prefer to avoid unnecessary allocations if possible. You currently allocate a vector as soon as `Actions` is created, and that vector is discarded if you call `remove_line` or `replace_with_lines`. Instead you could represent the inner state with an enum like this:

```rust
enum InnerState {
    Removed,
    Original(String),
    Replaced(Vec<String>),
}
```

...and create the final vector in `take_lines`. This way, if the line is removed no allocation will be performed (`Vec::new()` doesn't allocate) and no unnecessary allocation will be performed when the lines are replaced.